### ISSUE TITLE
Can't call JS function added by addFunction from secondary thread

### ISSUE THREAD

**Issue by moondayyy** [emscripten-core/emscripten#21273]
**Without** using **thread**(worker) the codes were **working** smoothly. 
However, when I using threads, I cannot get the pointer address of the `Module.addFunction` function.

**JavaScript:**

```
function is_speech(b) {
     console.log(b);
}

let pointerIsSpeech = Module.addFunction(is_speech, "vi");
Module.ccall('InitializeIsSpeechWeb', 'void', ['number'], [pointerIsSpeech]);
```

**C++:**

```
std::thread([&isSpeech](){
    isSpeech=Process();
    cb_isspech((void*)isSpeech);
}).detach();
```

I'm having the same problem as this one(#16567). 
mergeInto was mentioned. I tried, but I cannot access the function I defined (I think because I am inside the worker) 

**merge.js:**

```
mergeInto(LibraryManager.library,{
    cb_isspech: function (isSpeech){
       is_speech(isSpeech);  <-- It does not see the callback function I created.
       console.log("isSpeech: "+isSpeech);
    }
});
```

**How can I use JS callback function in MultiThread?**

**Comment by sbc100** [comment]
Yes, you because JS function are not shared between threads this doesn't currently work.   

We should probably just remove `addFunction` when threads are enabled since you are not the first person to be bitten by this.
 
The only JS code that you can call from your work is JS code that is part of `--library-js` or `--pre-js` or `--post-js`.   Other functions that you include in your main page are not visible to the worker.   So if you want your JS to be callable you need include it in the module itself (e.g. via `--library-js` or `--pre-js` or `--post-js`).  

Alternatively, you could proxy work back the main thread, where the rest of your code is.
